src/main/java/insat/company/platform/service/impl/ClubServiceImpl.java

```diff
@@ -111,4 +130,70 @@ public void delete(Long id) {
             .stream(clubSearchRepository.search(queryStringQuery(query)).spliterator(), false)
             .collect(Collectors.toList());
     }
+
+    @Override
+    public JoinClubRequest sendClubJoinRequest(Club club, User user) {
+        if (!(club.hasMember(user))) {
+            return Optional.ofNullable(joinClubRequestRepository.findOneByUserAndClubAndStatus(user, club, Status.PENDING))
+                .map((obj) -> {
+                    if (obj.isPresent()) {
+                        return null;
+                    }
+                    JoinClubRequest joinClubRequest = new JoinClubRequest();
+                    joinClubRequest.setClub(club);
+                    joinClubRequest.setUser(user);
+                    joinClubRequest.setRequestTime(LocalDate.now());
+                    joinClubRequest.setStatus(Status.PENDING);
+                    joinClubRequest = joinClubRequestRepository.save(joinClubRequest);
+                    log.info("{} {} Request Created  {} successfully !", joinClubRequest.getUser().getId(), joinClubRequest.getClub().getId());
+                    return joinClubRequest;
+                })
+                .orElseGet(() -> {
+                    JoinClubRequest joinClubRequest = new JoinClubRequest();
+                    joinClubRequest.setId(-1L);
```

Comment:
why are you setting the id to -1?

Reply:
we set the id to -1 because we couldn't return a null object so we created a join request with an id equal to -1 knowing that IDs starts from 1, in ClubResource, if the joinRequest's id is -1 none was created

Reply:
We are actually not saving it in the DB , it's only returned to the Ressource class to prepare the response Entity